tests/failpoints/cases/test_coprocessor.rs

```diff
@@ -172,3 +174,39 @@ fn test_region_error_in_scan() {
             .contains("region seek error")
     );
 }
+
+#[test]
+fn test_paging_scan() {
+    let data = vec![
+        (1, Some("name:0"), 2),
+        (2, Some("name:4"), 3),
+        (4, Some("name:3"), 1),
+        (5, Some("name:1"), 4),
+    ];
+
+    let product = ProductTable::new();
+    let (_, endpoint) = init_with_data(&product, &data);
+    // set batch size and grow size to 1, so that only 1 row will be scanned in each batch.
+    fail::cfg("copr_batch_initial_size", "return(1)").unwrap();
+    fail::cfg("copr_batch_grow_size", "return(1)").unwrap();
+    for paging_size in 1..=4 {
+        let req = DAGSelect::from(&product)
+            .paging_size(paging_size as u64)
+            .build();
+        let mut resp = handle_select(&endpoint, req);
```

Comment:
Should also check range in the response.

Reply:
It seems not easy to check it accurately, I checked its boundaries.